### PR TITLE
Add ranged value for complications

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Add ranged value for complications

#### WHY

A a succinct API for Ranged Values.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

**Comment by yschimke** [comment]
Not required, just noticed it missing.

**Comment by yschimke** [comment]
@ataulm going to land, happy to fix with any review fixes.

**Comment by LouisCAD** [comment]
There's a link between Tiles, Watch Face complications, and Horologist? I'm not sure I get it 🤔

**Comment by ataulm** [comment]
Horologist provides some helpers for tiles and helpers for complications (there's no link between tiles and complications).

File: tiles/src/main/java/com/google/android/horologist/tiles/complication/ComplicationTemplate.kt [hunk 1]
```diff
@@ -19,10 +19,12 @@ package com.google.android.horologist.tiles.complication
 import android.content.Context
 import androidx.wear.watchface.complications.data.ComplicationData
 import androidx.wear.watchface.complications.data.ComplicationType
+import com.google.android.horologist.tiles.ExperimentalHorologistTilesApi
 
 /**
  * A complication renderer from Data to any of the supported types.
  */
+@ExperimentalHorologistTilesApi
 public interface ComplicationTemplate<D> {
```
Comment (ataulm) [inline tiles/src/main/java/com/google/android/horologist/tiles/complication/ComplicationTemplate.kt:28]: @yschimke maybe @LouisCAD is referring to the package name, in which case +1, we should move

Comment (LouisCAD) [reply tiles/src/main/java/com/google/android/horologist/tiles/complication/ComplicationTemplate.kt:28]: Yes, the fact that it's in a tiles artifact, in a tiles package and, and has a `@ExperimentalHorologistTilesApi` annotation makes it very confusing, and looks like a mistake.

**Pull request merged**